### PR TITLE
use phylogenetic placement output in qiime

### PR DESCRIPTION
**Pull request by d4straub**
Closes https://github.com/nf-core/ampliseq/issues/561

Taxonomic tree and taxonomic annotation from phylogenetic placement are now picked up and used in downstream (QIIME2) analysis. 

The additional code in `FORMAT_PPLACETAX` uses only the taxonomic annotation from Gappa that
```
- has the lowest LWR, 
- if multiple, 
     - if identical it will choose the first one, 
     - if not identical, it will choose the one with less taxonomic entries, 
               - if same number of taxonomic entries, it will remove entries until they are identical, 
               - if all removed it will add a `NA`. 
```
I searched, tested and dismissed several pre-made code/commands/functions for this, such as from R's stringi or stringr packages, so I made the custom code.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/test.config [hunk 1]
```diff
@@ -38,8 +38,8 @@ params {
     metadata_category_barplot = "treatment1,badpairwise10"
 
     //restrict ANCOM analysis to higher taxonomic levels
-    dada_tax_agglom_max = 4
-    qiime_tax_agglom_max = 4
+    tax_agglom_max = 4
+    tax_agglom_max = 4
```
Comment (tillenglert) [inline conf/test.config:42]: This seems to be a copy paste error?

File: docs/output.md [hunk 2]
```diff
@@ -242,6 +242,7 @@ Phylogenetic placement grafts sequences onto a phylogenetic reference tree and o
 - `pplace/`
   - `*.graft.*.epa_result.newick`: Full phylogeny with query sequences grafted on to the reference phylogeny, in newick format.
   - `*.taxonomy.per_query.tsv`: Tab separated file with taxonomy information per query from classification by `gappa examine examinassign`
+  - `*.per_query_unique.tsv`: Tab separated file with taxonomy information as aboe, but one row per query, by filtering for lowest LWR (likelihood weight ratio)
```
Comment (tillenglert) [inline docs/output.md:245]:
```suggestion
  - `*.per_query_unique.tsv`: Tab separated file with taxonomy information as above, but one row per query, by filtering for lowest LWR (likelihood weight ratio)
```

File: modules/local/format_pplacetax.nf [hunk 3]
```diff
@@ -0,0 +1,96 @@
+process FORMAT_PPLACETAX {
+    tag "${tax.baseName}"
+    label 'process_high'
+
+    conda "bioconda::bioconductor-dada2=1.22.0 conda-forge::r-digest=0.6.30"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bioconductor-dada2:1.22.0--r41h399db7b_0' :
+        'quay.io/biocontainers/bioconductor-dada2:1.22.0--r41h399db7b_0' }"
```
Comment (tillenglert) [inline modules/local/format_pplacetax.nf:8]: Is dada2 needed for this module?

File: workflows/ampliseq.nf [hunk 4]
```diff
@@ -528,34 +526,44 @@ workflow AMPLISEQ {
...
         QIIME2_INASV ( ch_dada2_asv )
         QIIME2_INSEQ ( ch_fasta )
 
-        //Import taxonomic classification into QIIME2, if available
+        // Import phylogenetic tree into QIIME2
+        if ( params.pplace_tree ) {
+            ch_tree = QIIME2_INTREE ( FASTA_NEWICK_EPANG_GAPPA.out.grafted_phylogeny ).qza
+        } else { ch_tree = Channel.empty() }
+
+        // Import taxonomic classification into QIIME2, if available
```
Comment (tillenglert) [inline workflows/ampliseq.nf:538]: I would suggest to move the `tax_agglom_min = params.tax_agglom_min` and `tax_agglom_max = params.tax_agglom_max` assignments here and removing them in the statements below, apart from the cases they are changed.

**Review by tillenglert — APPROVED** [review]
Just some minor comments, but looks good to me!

File: conf/test.config (hunk 1 above)
Comment (d4straub) [reply conf/test.config:42]: An automated replacement error, yes, will fix!

File: workflows/ampliseq.nf (hunk 4 above)
Comment (d4straub) [reply workflows/ampliseq.nf:538]: Right, will do!

File: modules/local/format_pplacetax.nf (hunk 3 above)
Comment (d4straub) [reply modules/local/format_pplacetax.nf:8]: No, just base R. But the DADA2 container is used in any case in the pipeline, so it should save bandwidth and time to reuse it.

**Pull request merged**